### ISSUE TITLE
[Feature Request] Loading DLLs Similar to Revit Add-in Manager

### ISSUE THREAD

**Issue by johnpierson** [DynamoDS/Dynamo#9452]
## Dynamo version

Dynamo 1.x - 2.x

## Operating system

N/A

## What we currently see? 

When uninstalling or updating packages we have to completeley close Revit and Dynamo to finish the process. This is cumbersome and causing a lot of issues for users who may not know what the process is for updating packages with DLLs.

## What would I like to see?

I think it would be great if we could somehow refactor Dynamo to load DLLs in the same manner that the `Revit Add-in Manager` does. This tool is discussed on [the building coder](https://thebuildingcoder.typepad.com/blog/2011/05/debugging-an-add-in-without-restarting-revit.html) and on [archi-lab](http://archi-lab.net/debugging-revit-add-ins/) as well as several other places. The tool essentially loads DLLs on the fly and allows for easy debugging of your Revit add-ins. I feel like there should be some way to make this possible with Dynamo for ZeroTouch packages and the like.

**Comment by Dewb** [comment]
Agreed 100%. Thanks @johnpierson for starting the conversation here.

**Comment by mjkkirschner** [comment]
I think it's a bit more complicated since we'll need to dump the VM class tables etc as well, but we do that anyway when a graph opens. @Dewb whats the magic that allows unloading of the assemblies?
